src/components/badge/badge.style.ts

```diff
@@ -44,6 +44,7 @@ const StyledBadge = styled.span.attrs(({ onClick }) => ({
   width: 22px;
   min-height: 22px;
   border: solid 2px transparent;
+  z-index: 2;
```

Comment:
Added this to fix issue found during QA: 
![image](https://github.com/user-attachments/assets/d146332c-7c08-484a-aa19-c091782c9d15)